src/storage/src/redis_streams.cc

```diff
@@ -0,0 +1,1146 @@
+  return Status::OK();
+}
+
+Status RedisStreams::ScanKeys(const std::string& pattern, std::vector<std::string>* keys) {
+  std::string key;
+  rocksdb::ReadOptions iterator_options;
+  const rocksdb::Snapshot* snapshot;
+  ScopeSnapshot ss(db_, &snapshot);
+  iterator_options.snapshot = snapshot;
+  iterator_options.fill_cache = false;
+
+  rocksdb::Iterator* iter = db_->NewIterator(iterator_options, handles_[0]);
+  for (iter->SeekToFirst(); iter->Valid(); iter->Next()) {
+    ParsedStreamMetaValue parsed_stream_meta_value(iter->value());
+    if (parsed_stream_meta_value.length() != 0) {
+      key = iter->key().ToString();
+      if (StringMatch(pattern.data(), pattern.size(), key.data(), key.size(), 0) != 0) {
+        keys->push_back(key);
+      }
+    }
+  }
+  delete iter;
+  return Status::OK();
+}
+
+Status RedisStreams::PKPatternMatchDel(const std::string& pattern, int32_t* ret) {
+  rocksdb::ReadOptions iterator_options;
+  const rocksdb::Snapshot* snapshot;
+  ScopeSnapshot ss(db_, &snapshot);
+  iterator_options.snapshot = snapshot;
+  iterator_options.fill_cache = false;
+
+  std::string key;
+  std::string meta_value;
+  int32_t total_delete = 0;
+  Status s;
+  rocksdb::WriteBatch batch;
+  rocksdb::Iterator* iter = db_->NewIterator(iterator_options, handles_[0]);
+  iter->SeekToFirst();
+  while (iter->Valid()) {
+    key = iter->key().ToString();
+    meta_value = iter->value().ToString();
+    StreamMetaValue stream_meta_value;
+    stream_meta_value.ParseFrom(meta_value);
+    if ((stream_meta_value.length() != 0) &&
+        (StringMatch(pattern.data(), pattern.size(), key.data(), key.size(), 0) != 0)) {
+      stream_meta_value.InitMetaValue();
+      batch.Put(handles_[0], key, stream_meta_value.value());
+    }
+    if (static_cast<size_t>(batch.Count()) >= BATCH_DELETE_LIMIT) {
+      s = db_->Write(default_write_options_, &batch);
+      if (s.ok()) {
+        total_delete += static_cast<int32_t>(batch.Count());
+        batch.Clear();
+      } else {
+        *ret = total_delete;
+        return s;
+      }
+    }
+    iter->Next();
+  }
+  if (batch.Count() != 0U) {
+    s = db_->Write(default_write_options_, &batch);
+    if (s.ok()) {
+      total_delete += static_cast<int32_t>(batch.Count());
+      batch.Clear();
+    }
+  }
+
+  *ret = total_delete;
+  return s;
+}
+
+Status RedisStreams::Del(const Slice& key) {
+  std::string meta_value;
+  ScopeRecordLock l(lock_mgr_, key);
+  Status s = db_->Get(default_read_options_, handles_[0], key, &meta_value);
+  if (s.ok()) {
+    StreamMetaValue stream_meta_value;
+    stream_meta_value.ParseFrom(meta_value);
+    if (stream_meta_value.length() == 0) {
+      return Status::NotFound();
+    } else {
+      uint32_t statistic = stream_meta_value.length();
+      stream_meta_value.InitMetaValue();
+      s = db_->Put(default_write_options_, handles_[0], key, stream_meta_value.value());
+      UpdateSpecificKeyStatistics(key.ToString(), statistic);
```

Comment:
这个往下会规划运行Storage::DoCompactRange，里面没有stream类型的判断，导致对应的CompactRange不会调用，请修一下
```suggestion
      UpdateSpecificKeyStatistics(key.ToString(), statistic);
```

Reply:
ok，KEYS * 的单测也加了

Reply:
好的